internal/csi-addons/rbd/replication.go

```diff
@@ -497,6 +510,17 @@ func (rs *ReplicationServer) DemoteVolume(ctx context.Context,
 
 	// demote image to secondary
 	if mirroringInfo.Primary {
+		// store the image creation time for resync
+		_, err = rbdVol.GetMetadata(imageCreationTimeKey)
+		if err != nil && errors.Is(err, librbd.ErrNotFound) {
+			err = rbdVol.SetMetadata(imageCreationTimeKey, creationTime.String())
```

Comment:
Consider using your own `timestampToString()` and `timestampFromString()` functions. There is no guarantee that `creationTime.String()` produces the same string format across package versions.